exec/jdbc/src/test/java/org/apache/drill/jdbc/PreparedStatementTest.java

```diff
@@ -462,4 +618,25 @@ public void testParamSettingWhenUnsupportedTypeSaysUnsupported() throws SQLExcep
     }
   }
 
+
+  // Sets the SystemMaxRows option
+  private void setSystemMaxRows(int sysValueToSet) throws SQLException {
```

Comment:
I'm not sure that the approach used for blocking the connection should be implemented in this method. I think unit tests should acquire and release the lock.

Reply:
The unit tests use this method to set and release the lock. It reduces the lock code being repeated across unit tests.

Reply:
It looks like a hack to release the lock for one value and acquire for others. Except adding locks into tests directly, the more clear way may be to add a boolean flag into method argument whether to acquire or release the lock.

Also, I'm wondering whether these tests which change autoLimit does not affect other tests, i.e. when one test sets autoLimit to 3, but another test (which does not contain `setSystemMaxRows()` call, so it is not locked) expects result with 5 rows.

Reply:
There are _getter-setter_ method tests that don't rely on what the `SYSTEM` value is, so we don't need to acquire explicit locks. Only when a query is being executed, do we need to ensure that changes to the `SYSTEM` value are synchronized because that is when `QueryContext` will apply a `SESSION` or `SYSTEM` value.  Using the lock to synchronize the change, indirectly synchronizes these specific tests as well.
We don't have any other control on the parallelism or order of the tests within the 2 suites (`PreparedStatementTest` and `StatementTest`)... so using a common lock across both tests is the safest way to run such tests with an element of randomness (using the `RANDOMIZER` object).

Reply:
`PreparedStatementTest` contains `testSetQueryTimeoutAsZero()` test which verifies that the result should have 3 rows. This test will be executed without considering the `Semaphore` lock. So for the case when one of your tests changed the value of autoLimit, and before resetting the value this test is executed, it will return fewer rows than it should.

So adding these tests may introduce random failures of other tests.

Reply:
Strangely enough, these tests have never failed. I can set the `assert` to check for a non-zero value. Alternatively, I could disable the tests:
`testSetMaxRowsHigherThanSystemLimit()` and `testSetMaxRowsLowerThanSystemLimit()`

Reply:
As per our chat, I've introduced `@Before` and `@After` methods for synchronizing the `system` level modifications to the options.